sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/BlobPartitionManager.cs

```diff
@@ -58,42 +51,34 @@ public sealed class BlobPartitionManager : PartitionManager
         {
             List<PartitionOwnership> ownershipList = new List<PartitionOwnership>();
-
-            var prefix = $"{ fullyQualifiedNamespace }/{ eventHubName }/{ consumerGroup }/";
-            await foreach (BlobItem blob in _containerClient.GetBlobsAsync(traits: BlobTraits.Metadata, prefix: prefix).ConfigureAwait(false))
+            try
             {
-                // In case this key does not exist, ownerIdentifier is set to null.  This will force the PartitionOwnership constructor
-                // to throw an exception.
-
-                blob.Metadata.TryGetValue(BlobMetadataKey.OwnerIdentifier, out var ownerIdentifier);
-
-                long? offset = null;
-                long? sequenceNumber = null;
-
-                if (blob.Metadata.TryGetValue(BlobMetadataKey.Offset, out var str) && long.TryParse(str, out var result))
+                var prefix = $"{ fullyQualifiedNamespace }/{ eventHubName }/{ consumerGroup }/ownership/";
+                await foreach (BlobItem blob in _containerClient.GetBlobsAsync(traits: BlobTraits.Metadata, prefix: prefix).ConfigureAwait(false))
                 {
-                    offset = result;
+                    // In case this key does not exist, ownerIdentifier is set to null.  This will force the PartitionOwnership constructor
+                    // to throw an exception.
+
+                    blob.Metadata.TryGetValue(BlobMetadataKey.OwnerIdentifier, out var ownerIdentifier);
+
+                    ownershipList.Add(new InnerPartitionOwnership(
+                        fullyQualifiedNamespace,
+                        eventHubName,
+                        consumerGroup,
+                        ownerIdentifier,
+                        blob.Name.Substring(prefix.Length),
+                        blob.Properties.LastModified,
+                        blob.Properties.ETag.ToString()
+                    ));
                 }
 
-                if (blob.Metadata.TryGetValue(BlobMetadataKey.SequenceNumber, out str) && long.TryParse(str, out result))
-                {
-                    sequenceNumber = result;
-                }
-
-                ownershipList.Add(new InnerPartitionOwnership(
-                    fullyQualifiedNamespace,
-                    eventHubName,
-                    consumerGroup,
-                    ownerIdentifier,
-                    blob.Name.Substring(prefix.Length),
-                    offset,
-                    sequenceNumber,
-                    blob.Properties.LastModified,
-                    blob.Properties.ETag.ToString()
-                ));
+                return ownershipList;
+            }
+            catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.ContainerNotFound || ex.ErrorCode == BlobErrorCode.BlobNotFound)
+            {
+                throw new RequestFailedException("The specified container or blob does not exist: ", ex);
```

Comment:
For the time being we are throwing `RequestFailedException` exception when we get `ContainerNotFound ` or `BlobNotFound` error. For this scenario we need to decide the proper exception.

@jsquire let me know what do you think about the exception.

Reply:
Regarding the `BlobNotFound`, I think we won't get that. We are not searching for a specific blob name that might not exist. We are searching for blobs with a specified prefix and, if there are none, we'll just get an empty list. Correct me if I'm wrong.

Regarding `ContainerNotFound`, the only scenario I can think about is when someone deletes it manually. I think:
- We should not create a new one when starting if it does not exist. We'll eventually get the `ContainerNotFound` exception.
- There are three things we could do if we can't find (it was deleted or it never existed in the first place):
  - Crash everything. Let it surface and crash the processor.
  - Surface it to the processor, but gracefully delegate it to the Error Handler.
  - Try to recover and create it.

I'm between the two first options. I think we shouldn't stop processing because of that. In this scenario, we are using the same container for ownership and checkpoints, so in the end we can't keep processing. However, in a future scenario where different containers are used for them, we could keep the processor alive without being able to checkpoint (which is not exactly a good scenario, but we won't be losing events). However, I believe that if we just delegate it to the error handler, people will probably just ignore it and take some time to realize there's a problem.